.github/workflows/test.yml

```diff
@@ -66,14 +58,46 @@ jobs:
         uses: actions/cache@v2
         with:
           path: .tox
-          key: tox-cache-${{ matrix.python-version }}-${{ matrix.package }}-${{ matrix.os }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}-core
+          key: tox-cache-${{ env.RUN_MATRIX_COMBINATION }}-${{ hashFiles('tox.ini', 'dev-requirements.txt') }}-core
       - name: run tox
-        run: tox -f ${{ matrix.python-version }}-${{ matrix.package }}
+        run: tox -f ${{ matrix.python-version }}-${{ matrix.package }} -- --benchmark-json=${{ env.RUN_MATRIX_COMBINATION }}-benchmark.json
+      - name: Find and merge benchmarks
+        # TODO: Add at least one benchmark to every package type to remove this
+        if: matrix.package == 'core'
+        run: jq -s '.[0].benchmarks = ([.[].benchmarks] | add) | if .[0].benchmarks == null then null else .[0] end' opentelemetry-*/tests/*${{ matrix.package }}*.json > output.json
+      - name: Report on benchmark results
+        # TODO: Add at least one benchmark to every package type to remove this
+        if: matrix.package == 'core'
+        uses: rhysd/github-action-benchmark@v1
+        with:
+          name: OpenTelemetry Python Benchmark
+          tool: pytest
+          output-file-path: output.json
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          # Alert with a commit comment on possible performance regression
+          alert-threshold: 200%
+          comment-always: true
+          fail-on-alert: true
+          alert-comment-cc-users: '@${{ github.event.issue.user.login }}'
+      - name: Push benchmark result
```

Comment:
I'm slightly concerned that this means all the workflows will try to push to the `gh-pages` branch at the same time. Is this tool really smart enough to not push until all of the workflows across the matrix combinations finish?

From the [github-action-benchmarks docs](https://github.com/rhysd/github-action-benchmark#how-to-use):

> This action merges all benchmark results into one GitHub pages branch. If your workflows have multiple steps to check benchmarks from multiple tools, please give name input to each step to make each benchmark results identical.

Maybe it only applies to a single run, not across workflows?

Reply:
What does the format of the stored data in there look like out of curiosity? Link?

Git handles contention like this so it would get an error that their branch is out of date. And then it could retry?

Reply:
They won't as long as I update the namespace! https://nathanielrn.github.io/github-benchmarks-example/benchmarks/index.html